### DISCUSSION TITLE
Primeface CommandLink Not Working Inside UI:Repeat

### DISCUSSION THREAD

**Discussion by Santhanam12345** [primefaces/community#1771] [PrimeFaces] [answered]
Hi,

     My Project Specification is Primeface 12, Jakarta Faces 4.0, JDK21 and Tomcat 10.1

     My issue is p:commandlink action is not working inside UI:repeat for the first click. When i click the second time then the action is calling to bean. I checked the log files and i did not get any exception. Due to this issue i just replaced primefaces p:dataList and this one works. We have UI:repeat uses in many  files and its hard to replace with p:dataList.  I could not find the reason. If anyone facing this issue or found the solution kindly let me know. Thanks in advance.

**Comment by Rapster** [comment]
Did you try with `<h:commandLink />` ? If same problem, then the problem will be from JSF impl

**Comment by Santhanam12345** [comment]
Yes i tried. I got same issue . For first click it doesn't work but for second click it works.

**Reply by Rapster** [reply]
Not a PF problem then.

Report this issue to MyFaces or Mojarra team (depending what JSF impl you're using), ideally with a reproducer.

**Reply by Rapster** [reply]
FYI, I just tried and it works in both impl. Try to use latest version of MyFaces or Mojarra

**Comment by Santhanam12345** [comment]
I am using Jakarta Faces 4.0 and primefaces-12.0.0- jakarta

**Reply by Rapster** [reply, marked as answer]
Try to make a reproducer https://github.com/primefaces/primefaces-test
